includes/core/server/request/class-list-transactions.php

```diff
@@ -0,0 +1,279 @@
+	 *
+	 * @return Paginated|self
+	 */
+	public static function from_rest_request( $request ) {
+		$wcpay_request       = parent::from_rest_request( $request );
+		$date_between_filter = $request->get_param( 'date_between' );
+		$user_timezone       = $request->get_param( 'user_timezone' );
+
+		if ( ! is_null( $date_between_filter ) ) {
+			$date_between_filter = array_map(
+				function ( $transaction_date ) use ( $user_timezone, $wcpay_request ) {
+					return $wcpay_request->format_transaction_date_with_timestamp( $transaction_date, $user_timezone );
+				},
+				$date_between_filter
+			);
+		}
+
+		$filters = [
+			'match'                    => $request->get_param( 'match' ),
+			'date_before'              => $wcpay_request->format_transaction_date_with_timestamp( $request->get_param( 'date_before' ), $user_timezone ),
+			'date_after'               => $wcpay_request->format_transaction_date_with_timestamp( $request->get_param( 'date_after' ), $user_timezone ),
+			'date_between'             => $date_between_filter,
+			'type_is'                  => $request->get_param( 'type_is' ),
+			'type_is_not'              => $request->get_param( 'type_is_not' ),
+			'store_currency_is'        => $request->get_param( 'store_currency_is' ),
+			'customer_currency_is'     => $request->get_param( 'customer_currency_is' ),
+			'customer_currency_is_not' => $request->get_param( 'customer_currency_is_not' ),
+			'loan_id_is'               => $request->get_param( 'loan_id_is' ),
+			'search'                   => $request->get_param( 'search' ),
+		];
+
+		$wcpay_request->set_filters( $filters );
+		return $wcpay_request;
+	}
+
+	/**
+	 * Set match.
+	 *
+	 * @param string $match Match.
+	 *
+	 * @return void
+	 */
+	public function set_match( string $match ) {
+		$this->set_param( 'match', $match );
+	}
+
+	/**
+	 * Set loan id is.
+	 *
+	 * @param string $loan_id Loan id.
+	 *
+	 * @return void
+	 */
+	public function set_loan_id_is( string $loan_id ) {
+		$this->set_param( 'loan_id_is', $loan_id );
+	}
+
+	/**
+	 * Set currency is.
+	 *
+	 * @param string $currency Store currency.
+	 *
+	 * @return void
+	 */
+	public function set_store_currency_is( string $currency ) {
+		$this->set_param( 'store_currency_is', $currency );
+	}
+
+	/**
+	 * Set customer currency is.
+	 *
+	 * @param string $customer_currency_is Store currency.
+	 *
+	 * @return void
+	 */
+	public function set_customer_currency_is( string $customer_currency_is ) {
+		$this->set_param( 'customer_currency_is', $customer_currency_is );
+	}
+
+	/**
+	 * Set customer currency is not.
+	 *
+	 * @param string $currency Store currency.
+	 *
+	 * @return void
+	 */
+	public function set_customer_currency_is_not( string $currency ) {
+		$this->set_param( 'customer_currency_is_not', $currency );
+	}
+
+	/**
+	 * Set search.
+	 *
+	 * @param string $search Search term.
+	 *
+	 * @return void
+	 */
+	public function set_search( string $search ) {
+		$this->set_param( 'search', $search );
```

Comment:
This one is incorrect. If you take a look on the server, you'll see that `search` should be an array: https://github.com/Automattic/woocommerce-payments-server/blob/ce757b31f600ba0d2ee582978682856bf24031ce/server/wp-content/rest-api-plugins/endpoints/wcpay/class-transactions-controller.php#L88-L92

Also, there is a missing `WC_Payments_Utils::map_search_orders_to_charge_ids` call, which is otherwise present in the API client method: https://github.com/Automattic/woocommerce-payments/blob/7692800948ef80d3118c5593c32ee6776ea51022/includes/wc-payment-api/class-wc-payments-api-client.php#L666-L668

Reply:
This is fixed.